### ISSUE TITLE
Support for BatchGetItem

### ISSUE THREAD

**Issue by yyamanoi1222** [aws/aws-sdk-ruby-record#122]
Are there any plans to implement BatchGetItem in this gem?

I saw this issue https://github.com/aws/aws-sdk-ruby-record/issues/12
We have a requirement and BatchGetItem may be better than TransactGetItems. (e.g., retrieve large number of items , no need to consider transaction conflict errors, etc.)

If you don't plan to implement it and are be open the idea, may I add the feature myself?
The API interface will be something like transact_find

```ruby
result = Aws::Record::Batch.find(
  batch_items: [
    TableOne.bfind_opts(key: { uuid: "uuid1234" }),
    TableTwo.tfind_opts(key: { hk: "hk1", rk: "rk1"}), 
    TableTwo.tfind_opts(key: { hk: "hk2", rk: "rk2"})  
  ]
)

result.responses # { TableOne.table_name => [TableOne], TableTwo.table_name => [TableTwo, TableTwo] }
```

Thanks!

**Comment by alextwoods** [comment]
Thanks for reaching out on this - we do have plans to add support for BatchGetItems!  As part of a new investment in this library we are planning to add support following the approach taken in [PR #119](https://github.com/aws/aws-sdk-ruby-record/pull/119) (https://github.com/aws/aws-sdk-ruby-record/pull/119) where we added support for batch writing items using a new [BatchWrite](https://github.com/aws/aws-sdk-ruby-record/blob/main/lib/aws-record/record/batch_write.rb) (https://github.com/aws/aws-sdk-ruby-record/blob/main/lib/aws-record/record/batch_write.rb) class to avoid any backwards compatibility / breaking changes.   

We are planning to add support for [batch_get_item](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/DynamoDB/Client.html#batch_get_item-instance_method) (https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/DynamoDB/Client.html#batch_get_item-instance_method) with a BatchRead class (capable of reading multiple items from multiple tables) which can be used by individual Record classes, giving an interface like:

```ruby
class Foo
  include Aws::Record
  string_attr :name, hash_key: true
end

class Bar
  include Aws::Record
  string_attr :id, hash_key: true
  string_attr :rk, range_key: true
end

# produces a heterogeneous ItemCollection
items = Aws::Record::Batch.read do |db|
  db.find(Foo, name: 'n1')
  db.find(Foo, name: 'n2')
  db.find(Bar, id: 'id1', rk: 'rk1')
  db.find(Bar, id: 'id1', rk: 'rk2')
end

# returns a homogenous ItemCollection
foo_items = Foo.find_all(
  [
    {name: 'n1'},
    {name: 'n2'}
  ])

# an alternative syntax
foo_items = Foo.find_all do |db|
  db.find(name: 'n1') # the Foo class is implied here
  db.find(name: 'n2')
end
```

Would that work for your use case?

**Comment by yyamanoi1222** [comment]
Thank you for your reply
It looks like good!

**Comment by yyamanoi1222** [comment]
@alextwoods 
What do you think about the above?

And when will this function be implemented?
I wonder if I should extend this gem to use it or wait for implementation.
I can always help with implementation about this function.
Sorry to bother you.

**Comment by alextwoods** [comment]
Sorry for the delay in response.  Yes, the existing `ItemCollection` class wouldn't quite work for this case - and we would return an enumerable with a similar interface (supporting an `each` method) rather than modifying the ItemCollection class itself.  

I believe we are currently targeting about mid Q4 of this year for the feature.

**Comment by jterapin** [comment]
Implemented the feature and published a new gem (v2.10.0).
